### ISSUE TITLE
t/bin.t is using wrong perl

### ISSUE THREAD

**Issue by eserte** [gryphonshafer/dest#15]
t/bin.t fails on most of my smoker systems:
```
#   Failed test 'help'
#   at t/bin.t line 9.
#                   ''
#     doesn't match '(?^:Usage:\s+dest COMMAND \[OPTIONS\])'

#   Failed test 'version'
#   at t/bin.t line 12.
#                   ''
#     doesn't match '(?^:^dest version [\d\.]+$)'
# Looks like you failed 2 tests of 2.
t/bin.t .................. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests 
```
Unfortunately there's no useful test diagnostics, so I run the test through strace. Here an interesting log line from this run:
```
28272 20:52:15.876317 write(2, "Can't locate App/Dest.pm in @INC (you may need to install the App::Dest module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.24.1 /usr/local/share/perl/5.24.1 /usr/lib/x86_64-linux-gnu/perl5/5.24 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.24 /usr/share/per
l/5.24 /usr/local/lib/site_perl /usr/lib/x86_64-linux-gnu/perl-base) at dest line 10.\nBEGIN failed--compilation aborted at dest line 10.\n", 420 <unfinished ...>
```
The `@INC` path looks like Debian's system perl (which is the first perl in user's PATH) --- but in this case the build was running with another self-compiled perl. Probably at some point you have to use `$^X` to point to the correct perl binary.

**Comment by gryphonshafer** [comment]
Ah, OK. I think I see what's going on. Maybe. [t/bin.t line 8](https://github.com/gryphonshafer/dest/blob/cd88818669928d66b06df2e00f2a0087f8c2e81e/t/bin.t#L8) and [line 11](https://github.com/gryphonshafer/dest/blob/cd88818669928d66b06df2e00f2a0087f8c2e81e/t/bin.t#L11) are calling Perl via:

    /usr/bin/env perl dest help

So yeah, probably need to use `$^X` instead. Unfortunately, I can't repro the error myself. Will try doing a deeper look later today.

**Comment by gryphonshafer** [comment]
Closed with b4e7fcc, probably. (I can't repro this locally. I'm just guessing this fixes things.)

**Comment by eserte** [comment]
I think the change looks good. Will see what the smokers say...
